vpp/abx/vpp2202/test/test_abx.py

```diff
@@ -0,0 +1,577 @@
+        return acl.acl_index
+
+    def create_stream(
+        self,
+        in_if,
+        out_if,
+        vlan_id=0,
+        ttl=64,
+    ):
+        """
+        Create input packet stream for defined interface.
+        If VLAN tag if other than 0.
+        """
+        pkts = []
+        if vlan_id == 0:
+            p = (
+                Ether(dst=in_if.local_mac, src=in_if.remote_mac)
+                / IP(dst=out_if.remote_ip4, src=in_if.remote_ip4, ttl=ttl)
+                / TCP(
+                    sport=3000,
+                    dport=3000,
+                )
+            )
+        else:
+            p = (
+                Ether(dst=in_if.local_mac, src=in_if.remote_mac)
+                / Dot1Q(vlan=vlan_id)
+                / IP(dst=out_if.remote_ip4, src=in_if.remote_ip4, ttl=ttl)
+                / TCP(
+                    sport=3000,
+                    dport=3000,
+                )
+            )
+
+        pkts.append(p)
+        return pkts
+
+    def create_apply_acl(self):
+        """
+        Create and apply ACL rule
+        Return ACL rule index
+        """
+
+        # Create ACL rule
+        rules = []
+        rules.append(
+            self.create_rule(
+                self.IPV4, self.PERMIT, 3000, self.proto[self.IP][self.TCP]
+            )
+        )
+
+        # Apply rules
+        return self.apply_rules(rules, "permit all")
+
+    def create_abx_policy(self, policy_id, acl_idx, dest_if, change_mac=True):
+        """
+        Create and add ABX policy
+        Returns ABX config as map.
+        """
+
+        policy = {
+            "policy_id": policy_id,
+            "acl_index": acl_idx,
+            "tx_sw_if_index": dest_if.sw_if_index,
+            "dst_mac": dest_if.local_mac,
+        }
+
+        if change_mac is False:
+            del policy["dst_mac"]
+
+        self.vapi.abx_policy_add_del(1, policy)
+        return policy
+
+    def attach_abx_policy(self, policy_id, rx_sw_if_index):
+        """
+        Attach ABX config to interface.
+        Returns ABX attach config.
+        """
+
+        interface_attach = {
+            "policy_id": policy_id,
+            "priority": 0,
+            "rx_sw_if_index": rx_sw_if_index,
+        }
+
+        self.vapi.abx_interface_attach_detach(is_attach=1, attach=interface_attach)
+        return interface_attach
+
+    def cleanup_abx(self, abx_policy_config, abx_attach_config):
+        """
+        Cleanup ABX attach config and policy.
+        """
+
+        self.vapi.abx_interface_attach_detach(is_attach=0, attach=abx_attach_config)
+        self.vapi.abx_policy_add_del(0, abx_policy_config)
+
+    def send_and_expect(
+        self,
+        intf,
+        pkts,
+        output,
+        n_rx=None,
+        worker=None,
+        trace=True,
+        msg=None,
+        stats_diff=None,
+    ):
+        return super().send_and_expect(
+            intf, pkts, output, n_rx, worker, trace, msg, stats_diff
```

Comment:
I have been getting following error. Removing `msg` and `stats_diff` parameters from method call fixed it.
```
Traceback (most recent call last):
  File "/home/maros/vpp/test/test_abx.py", line 277, in test_abx_basic_acl_mac_dst_sub_if
    pkts = self.send_and_expect(self.pg0, pkts, self.pg1)
  File "/home/maros/vpp/test/test_abx.py", line 256, in send_and_expect
    return super().send_and_expect(
TypeError: send_and_expect() takes from 4 to 7 positional arguments but 9 were given
```

Reply:
```send_and_expect``` method fixed as mentioned above.

For checkstyle, decision is to use one from ```v22.02``` branch.

IPv6 tests are not priority at the moment, listed as TODO. Negative scenario could be implemented, but for now should be handled by ACL test cases since they determine which packets are to be forwarded.